src/coreclr/vm/method.hpp

```diff
@@ -1927,10 +1940,21 @@ class MethodDesc
         LIMITED_METHOD_DAC_CONTRACT;
         if (!HasAsyncMethodData())
             return false;
-        auto asyncKind = GetAddrOfAsyncMethodData()->kind;
-        return asyncKind == AsyncMethodKind::AsyncVariantThunk ||
-            asyncKind == AsyncMethodKind::AsyncVariantImpl ||
-            asyncKind == AsyncMethodKind::AsyncExplicitImpl;
+
+        AsyncMethodKind asyncKind = GetAddrOfAsyncMethodData()->kind;
+        return hasAsyncKindFlags(asyncKind, AsyncMethodKind::Async);
+    }
+
+    // Is this a Variant method?
+    // If yes, the method has another variant.
+    inline bool IsVariantMethod() const
```

Comment:
I do not understand how this is different from IsAsyncVariant.

Also, "Variant" without "Async" it makes it look like it has something to do with type variance that is completely different feature.

Reply:
`Variant` flag only tells that this is one from a pair. Basically means we can call `GetAsyncOtherVariant` and get the other variant.

Perhaps `HasAsyncOtherVariant` is a better name. - kind of matches `GetAsyncOtherVariant`.

Reply:
> standardize on AsyncCall to mean async calling convention

If we were to standardize on AsyncCallConv to mean async calling convention, this would be `IsAsyncCallConvVariant` / `GetAsyncCallConvVariant`